host_core/lib/host_core/wasmcloud/runtime_server.ex

```diff
@@ -132,10 +132,11 @@ defmodule HostCore.WasmCloud.Runtime.Server do
       # This callback is invoked by the wasmcloud::Runtime's host call handler
       payload = payload |> IO.iodata_to_binary()
 
+      # Convert the `call_context` list of bytes to the `span_ctx` term
       with true <- is_list(call_context),
-           ctx_list <- :erlang.list_to_binary(call_context),
-           true <- is_binary(ctx_list),
-           span_ctx <- :erlang.binary_to_term(ctx_list) do
+           ctx_binary <- :erlang.list_to_binary(call_context),
+           true <- is_binary(ctx_binary),
```

Comment:
potentially superfluous question - doesn't `:erlang.list_to_binary` always return a bitstring value ? If so, do we need the `true <- is_binary(ctx_binary)` guard?

Reply:
https://www.erlang.org/doc/man/erlang.html#list_to_binary-1 this looks like it doesn't allow non-binary output

Reply:
I think so, I was really just being overly cautious

Reply:
These functions either work or panic so I wanted to ensure we were going down the proper path with the `with`

Reply:
ok it looks like this is "with safe" (e.g. no panic, just match fail).

```
** (ArgumentError) errors were found at the given arguments:

  * 1st argument: not an iolist term

    :erlang.list_to_binary(nil)
    iex:1: (file)
iex(1)> :erlang.list_to_binary([])
""
iex(2)> :erlang.list_to_binary([65, 66, 67])
"ABC"
iex(3)>
```